File: java/src/test/java/problems1_100/MaximumSubarrayTest.java
package problems1_100;

import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class MaximumSubarrayTest extends SampledTest {
    @Test
    public void test() {
        for (MaximumSubarray solution : MaximumSubarray.values()) {
            for (MaximumSubarraySample sample : getSamples(MaximumSubarraySample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(solution.maxSubArray(sample.input.nums))
                    .isEqualTo(sample.output);
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems1_100/MaximumSubarrayTest.kt
package problems1_100

import com.google.common.truth.Truth.assertWithMessage
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import sample.getSamples
import kotlin.test.Test

class MaximumSubarrayTest : SampledTest() {
    @Test
    fun test() =
        MaximumSubarray.entries.forEach { solution ->
            getSamples<Array<MaximumSubarraySample>>().forEach { (input, output, message) ->
                assertWithMessage(message)
                    .that(solution.maxSubArray(input.nums))
                    .isEqualTo(output)
            }
        }
}
